code/game/objects/structures/barsign.dm

```diff
@@ -97,11 +98,10 @@ var/list/barsigns = list()
 	return attack_hand(user)
 
 /obj/structure/sign/double/barsign/attack_ghost(mob/user)
-	if(isAdminGhost(user))
-		attack_hand(user)
+	attack_hand(user)
 
 /obj/structure/sign/double/barsign/attack_hand(mob/user)
-	if(!isAdminGhost(user))
+	if(!isAdminGhost(user) || !isAI(user))
 		if(!allowed(user))
```

Comment:
can simplify to one line `if(!(isAdminGhost(user) || !isAI(user))&&!allowed(user))

Reply:
Got it